kernel/aster-nix/src/process/signal/sig_mask.rs

```diff
@@ -81,4 +131,72 @@ impl SigMask {
         let idx = Self::num_to_idx(signum);
         self.bits |= 1_u64 << idx;
     }
+
+    fn num_to_idx(num: SigNum) -> usize {
+        (num.as_u8() - MIN_STD_SIG_NUM) as usize
+    }
+}
+
+// This is to allow hexadecimally formatting a `SigSet` when debug printing it.
+impl LowerHex for SigSet {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        LowerHex::fmt(&self.bits, f) // delegate to u64's implementation
+    }
+}
+
+/// An atomic signal mask.
+///
+/// All operations to this signal uses the [`Relaxed`] ordering. So the precise
+/// order of blocking and unblocking signals may not be consistent among
+/// threads. Blocking and unblocking signals cannot fence out any critical
+/// sections either.
+///
+/// [`Relaxed`]: core::sync::atomic::Ordering::Relaxed
+pub struct AtomicSigMask(AtomicU64);
```

Comment:
Again, I would like to suggest naming this type `AtomicSigSet` and having a `AtomicSigMask` as its alias. The rationale is that a collection type (such as `Vec` or `SigSet` or `AtomicSigSet`) should be decoupled from its usage. It could be used as a mask, but it could be used in some other way. Calling it a _set_ is more neutral to use cases. For example, we may in the future use this type as a field of `SigQueues` to track _pending_ signals. In this case, calling it `AtomicSigSet` obviously conveys the intention better than `AtomicSigMask`.